### ISSUE TITLE
How to Load Second SQL Table and Use IF Statements For It?

### ISSUE THREAD

**Issue by thadius856** [RaspberryPints/RaspberryPints#32]
I created a second table succesfully, `config`. This is what mine looks like (snippet from a .sql):

```
--
-- Table structure for table `config`
--

CREATE TABLE IF NOT EXISTS `config` (
  `configid` int(11) NOT NULL AUTO_INCREMENT,
  `showTapNumCol` tinyint(1) NOT NULL,
  `showSrmCol` tinyint(1) NOT NULL,
  `showIbuCol` tinyint(1) NOT NULL,
  `showCalCol` tinyint(1) NOT NULL,
  `showKegCol` tinyint(1) NOT NULL,

  PRIMARY KEY (`configid`)
) ENGINE=InnoDB  DEFAULT CHARSET=latin1 AUTO_INCREMENT=4 ;

--
-- Dumping data for table `config`
--

INSERT INTO `config` (`configid`, `showTapNumCol`, `showSrmCol`, `showIbuCol`, `showCalCol`, `showKegCol`) VALUES
(1, 1, 1, 1, 1, 1);
```

The table is successfully created. However, when I try to load a second table in the PHP section before the head in index.php, I get wonky errors.

What is the correct way to load it? :\

**Comment by JayUnt** [comment]
One this that WordPress does with their config table is not have columns for each config, but to have just 3 columns in the table, id, option_name and option_value.  Then if we were to add more config properties, we just need to insert new rows into the db and not alter the table.   

We could have a php file that stores all of the option_name that we use in an array just like an enum.  That way we dont hardcode the names in all of the spots, That is a better practice.

As for calling two inserts on the page, it works for me.  i ran your table script to get the table, and then in the if($db), right after the db() call, i put this code:

```
$config = array();
$sql = "SELECT * FROM config WHERE configid = 1";
$qry = mysql_query($sql);
while($b = mysql_fetch_array($qry))
{
    array_push($config, $b);
}
```

Then right after the body tag i put this code, and it outputted all of the properties and their values.

```
<?php print_r($config); ?>
```

**Comment by JayUnt** [comment]
also configid should really be just id.  We are in the config table, so we know what the id is for.  I think of each table like it would be a model.  If you had a config object, you would just want to get its id by saying 

```
config.id
```

and not 

```
config.configId
```

**Comment by thadius856** [comment]
> One this that WordPress does with their config table is not have columns for each config, but to have just 3 columns in the table, id, option_name and option_value. Then if we were to add more config properties, we just need to insert new rows into the db and not alter the table.
> 
> We could have a php file that stores all of the option_name that we use in an array just like an enum. That way we dont hardcode the names in all of the spots, That is a better practice.

Wow. Hrm. I think I get what you're saying. But I have no idea how to make that happen.

> As for calling two inserts on the page, it works for me. i ran your table script to get the table, and then in the if($db), right after the db() call, i put this code:

Little lost here. Can you paste everything before the html open tag?

**Comment by thadius856** [comment]
> also configid should really be just id. We are in the config table, so we know what the id is for.

Hrm. K. I'll keep that in mind.

**Comment by JayUnt** [comment]
here you go

```
<?php
    require("includes/config.php");

    //This can be used to choose between CSV or MYSQL DB
    $db = true;

    // Setup array for all the beers that will be contained in the list
    $beers = array();

    if($db){
        // Connect to the database
        db();


        $config = array();
        $sql = "SELECT * FROM config WHERE configid = 1";
        $qry = mysql_query($sql);
        while($b = mysql_fetch_array($qry))
        {
            array_push($config, $b);
        }

        $sql = "SELECT * FROM beers WHERE active = true ORDER BY tapnumber";
        $qry = mysql_query($sql);
        while($b = mysql_fetch_array($qry))
        {
            $beeritem = array(
                "id" => $b['beerid'],
                "beername" => $b['name'],
                "style" => $b['style'],
                "notes" => $b['notes'],
                "og" => $b['og'],
                "fg" => $b['fg'],
                "srm" => $b['srm'],
                "ibu" => $b['ibu'],
                "kegstart" => $b['kegstart'],
                "kegremain" => $b['kegremain'],
                "tapnumber" => $b['tapnumber'],
            );
            array_push($beers, $beeritem);
        }
    } else {

    }
?>
```

**Comment by JayUnt** [comment]
And for the actual table structure for the config, this is what we could use.  I mostly stole it from WP.

```
use raspberrypints;

--
-- Table structure for table `config`
--

CREATE TABLE IF NOT EXISTS `config` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `config_name` varchar(50) NOT NULL,
  `config_value` longtext NOT NULL,

  PRIMARY KEY (`id`)
) ENGINE=InnoDB  DEFAULT CHARSET=latin1 AUTO_INCREMENT=4 ;

--
-- Dumping data for table `config`
--

INSERT INTO `config` (config_name, config_value) VALUES ('showTapNumCol', 1);
INSERT INTO `config` (config_name, config_value) VALUES ('showSrmCol', 1);
INSERT INTO `config` (config_name, config_value) VALUES ('showIbuCol', 1);
INSERT INTO `config` (config_name, config_value) VALUES ('showCalCol', 1);
INSERT INTO `config` (config_name, config_value) VALUES ('showKegCol', 1);
```

We would grab all of the config properties from the table and throw it into an array. like this:

```
$config = array();
$sql = "SELECT * FROM config";
$qry = mysql_query($sql);
while($b = mysql_fetch_array($qry))
{
    $config[$b['config_name']] = $b['config_value'];
}
```

then we could use $config['showTapNumCol'] to get if we should show it.  I was talking about an array to store all of the config_name values, but now looking into it more we really want to use a class with constants, like this page says http://stackoverflow.com/questions/254514/php-and-enumerations.  So we would call $config[ConfigName::ShowTapNumCol].  That way the actual string for the config_names are in a central location and not strings all over the site.

Make sense?

**Comment by thadius856** [comment]
> Make sense?

The first half. Then... n... no?

That said...

![make-it-so-captain](https://f.cloud.github.com/assets/6320180/1898025/6fb25f60-7c08-11e3-9bb2-d359eee31700.jpeg)

**Comment by JayUnt** [comment]
Haha. Sounds good. I'll work on it tomorrow.

**Comment by JayUnt** [comment]
I committed the config table, loading the config data and using the config to hide/show columns.  I also made a sql folder with the schema scripts, as well as took the sample data out of the file and made it its own script.  This way when we are to push it, we wont be shoving 10 beers into their tables, but we still have it for testing.

I didn't change the beers table at all (besides moving it to a new file), just the config stuff.  I was thinking about creating the other tables but then i thought skemp might have been doing that.
